### PR TITLE
feat(v1): Init vote extension handlers

### PR DESCRIPTION
**Pull request by davidterpay**
## Overview
This PR initializes the vote extension handlers as a part of `ABCIHandler` (formerly `ProposalHandler`). Shuffled some files around so that things are a bit cleaner as well.

### CODE REVIEW

File: abci/abci.go [hunk 1]
```diff
@@ -23,7 +19,8 @@ type (
 		IsAuctionTx(tx sdk.Tx) (bool, error)
 	}
 
-	ProposalHandler struct {
+	//nolint:all
+	ABCIHandler struct {
```
Comment (davidterpay) [inline abci/abci.go:23]: Added this here for purposes of keeping the name `ABCIHandler` since go lint was not happy.

Comment (alexanderbez) [reply abci/abci.go:23]: We should have two handlers:

* `ProposalHandler`
* `VoteExtensionHandler`

File: abci/proposals.go
Comment (alexanderbez) [inline abci/proposals.go:1]: We're going to need to ask ourselves if we still want to support v0 of POB (what's currently under `release/v1.x.x`), i.e. for chains that want to use POB for 0.47.x. If we want to support this, then we can't have the structure as -is. We'll need something like the following:

* `abci/v1/...`
* `abci/v2/...`

File: abci/vote_extensions.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkmempool "github.com/cosmos/cosmos-sdk/types/mempool"
+)
+
+type (
+	// MempoolVoteExtensionI contains the methods required by the VoteExtensionHandler
+	// to interact with the local mempool.
+	MempoolVoteExtensionI interface {
```
Comment (alexanderbez) [inline abci/vote_extensions.go:13]:
```suggestion
	VoteExtensionMempool interface {
```

File: abci/vote_extensions.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		Remove(tx sdk.Tx) error
+		AuctionBidSelect(ctx context.Context) sdkmempool.Iterator
+		IsAuctionTx(tx sdk.Tx) (bool, error)
+	}
+
+	VoteExtensionHandler struct {
```
Comment (alexanderbez) [inline abci/vote_extensions.go:19]: nit: godoc

**Review by alexanderbez — APPROVED** [review]

**Pull request merged**